src/main/java/com/yahoo/memory/Memory.java

```diff
@@ -111,20 +111,33 @@ public static Memory wrap(final boolean[] arr) {
   }
 
   /**
-   * Wraps the given primitive array for read operations
+   * Wraps the given primitive array for read operations, with native byte order.
    * @param arr the given primitive array
    * @return Memory for read operations
    */
   public static Memory wrap(final byte[] arr) {
-    nullCheck(arr);
-    if (arr.length == 0) {
-      return WritableMemoryImpl.ZERO_SIZE_MEMORY;
-    }
-    return new WritableMemoryImpl(new ResourceState(arr, Prim.BYTE, arr.length));
+    return wrap(arr, ByteOrder.nativeOrder());
+  }
+
+  /**
+   * Wraps the given primitive array for read operations, with the given byte order.
+   * @param arr the given primitive array
+   * @param byteOrder the byte order
+   * @return Memory for read operations
+   */
```

Comment:
Please explain more what the intent is and how this will be used.  Even if you set `ByteOrder` to BigEndian none of our put/get methods will actually do any byte swapping.  So how do you intend to use this ByteOrder parameter, other than just setting the field in the ResourceState?  How does this help?

Reply:
Indeed, currently it does nothing. But we need big-endian memory in Druid, for deep compatibility reasons. I think we should add `NonNativeBOWritableMemoryImpl` and `NonNativeBOWritableBufferImpl` that does swapping. I can do this after this PR is merged.